test/tstCompileOnlyGeometry.cpp

```diff
@@ -218,6 +218,22 @@ void test_geometry_compile_only()
   // check_valid_geometry_traits(WrongCoordinateSpecialization{});
 }
 
+void test_point_cv_compile_only()
+{
+  using Point = ArborX::Point;
+  namespace GT = ArborX::GeometryTraits;
+
+  static_assert(GT::dimension_v<const Point> == 3);
+  static_assert(std::is_same_v<GT::coordinate_type_t<const Point>, float>);
+  static_assert(std::is_same_v<GT::tag_t<const Point>, GT::PointTag>);
+  static_assert(GT::is_point_v<const Point>);
+
+  static_assert(GT::dimension_v<volatile Point> == 3);
+  static_assert(std::is_same_v<GT::coordinate_type_t<volatile Point>, float>);
+  static_assert(std::is_same_v<GT::tag_t<volatile Point>, GT::PointTag>);
+  static_assert(GT::is_point_v<volatile Point>);
```

Comment:
```suggestion
```
I would not bother with `volatile`.
I also like being able to grep for it w/o false positives.